src/server/main_service.cc

```diff
@@ -1536,38 +1537,18 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
 
   CmdArgVec arg_vec, tmp_keys;
 
-  // We ignore transaction mode in case it's filled only with EVAL-like commands.
-  // This is done to support OptimalBits/bull js framework
-  // that for some reason uses MULTI to send multiple jobs via EVAL(SHA) commands,
-  // instead of using pipeline mode.
-  // TODO: to check with BullMQ developers if this is a bug or a feature.
-  if (state == ExecEvalState::ALL) {
-    string cmd_name;
-    auto body = move(exec_info.body);
-    exec_info.Clear();
-    Transaction* trans = cntx->transaction;
-    cntx->transaction = nullptr;
-
-    SinkReplyBuilder::ReplyAggregator agg(rb);
-    rb->StartArray(body.size());
-    for (auto& scmd : body) {
-      arg_vec.resize(scmd.NumArgs() + 1);
-      // We need to copy command name to the first argument.
-      cmd_name = scmd.Cid()->name();
-      arg_vec.front() = MutableSlice{cmd_name.data(), cmd_name.size()};
-      auto args = absl::MakeSpan(arg_vec);
-      scmd.Fill(args.subspan(1));
-
-      DispatchCommand(args, cntx);
-    }
-    cntx->transaction = trans;
+  // Check if script most LIKELY has global eval transactions
+  bool global_script = script_mgr()->AreGlobalByDefault();
+  int multi_mode = absl::GetFlag(FLAGS_multi_exec_mode);
 
-    return;
+  if (state != ExecEvalState::NONE) {
```

Comment:
so suppose `multi_mode` is lock ahead and lua is global. 
what will happen upon running lua? will it actually lock all the shards in addition to key locks we have in multi-transaction? will they be lifted as well?

Reply:
in other words, do we actually preserve the global property of lua script?

Reply:
Inside StartMultiExec there is the following lines which makes sure that if there is global command or a global script in the multi trasaction we will not use lock ahead bug gloabl lock
  // Atomic modes fall back to GLOBAL if they contain global commands.
  if (global && multi_mode == Transaction::LOCK_AHEAD)
    multi_mode = Transaction::GLOBAL;

Reply:
ok, please add comment below `// Allow multi eval only when scripts` that says that we adjust the atomicity level of multi transaction inside `StartMultiExec`